src/viperleed/calc/files/beams.py

```diff
@@ -239,11 +241,12 @@ def readOUTBEAMS(filename="EXPBEAMS.csv", sep=",", enrange=None):
             with open(filename, 'r') as rf:
                 lines = [li[:-1] for li in rf.readlines()]
         except FileNotFoundError:
-            if filename.endswith(".csv") and os.path.isfile(filename[:-4]):
-                with open(filename[:-4], 'r') as rf:
+            filename = Path(filename)
+            if filename.suffix == '.csv' and filename.with_suffix('').is_file:
+                with open(filename.with_suffix(''), 'r') as rf:
                     lines = [li[:-1] for li in rf.readlines()]
             else:
-                logger.error("Error reading "+filename)
+                logger.error(f"Error reading {str(filename)}.")
```

Comment:
```suggestion
                logger.error(f"Error reading {filename}.")
```

No need for `str`.

Reply:
Sorry, this somehow got lost. I've implemented the suggestions.